0013-libxl-stubdom-add-kernel-ramdisk-options.patch

```
@@ -0,0 +1,92 @@
+         }
+     }
+     def->emulator_cmdline = virXPathString("string(./devices/emulator/@cmdline)", ctxt);
++    def->emulator_kernel = virXPathString("string(./devices/emulator/@kernel)", ctxt);
++    def->emulator_ramdisk = virXPathString("string(./devices/emulator/@ramdisk)", ctxt);
+ 
+     n = virXPathULong("string(./devices/emulator/@memory)",
+                       ctxt,
+@@ -30063,6 +30065,8 @@ virDomainDefFormatInternalSetRootName(vi
+         if (def->emulator_memory != 0)
+             virBufferAsprintf(buf, " memory='%lu'", def->emulator_memory);
+         virBufferEscapeString(buf, " cmdline='%s'", def->emulator_cmdline);
++        virBufferEscapeString(buf, " kernel='%s'", def->emulator_kernel);
++        virBufferEscapeString(buf, " ramdisk='%s'", def->emulator_ramdisk);
+         if (!def->emulator) {
+             virBufferAddLit(buf, "/>\n");
+         } else {
+diff -bpru libvirt-6.6.0/src/conf/domain_conf.h libvirt-6.6.0.new/src/conf/domain_conf.h
+--- libvirt-6.6.0/src/conf/domain_conf.h	2020-12-23 19:05:01.963313040 +0300
++++ libvirt-6.6.0.new/src/conf/domain_conf.h	2020-12-23 18:59:44.817841019 +0300
+@@ -2570,6 +2570,8 @@ struct _virDomainDef {
+     char *emulator;
+     virDomainEmulatorType emulator_type;
+     char *emulator_cmdline;
++    char *emulator_kernel;
++    char *emulator_ramdisk;
+     unsigned long emulator_memory;
+     /* Most {caps_,hyperv_,kvm_,}feature options utilize a virTristateSwitch
+      * to handle support. A few assign specific data values to the option.
+diff -bpru libvirt-6.6.0/src/libxl/libxl_conf.c libvirt-6.6.0.new/src/libxl/libxl_conf.c
+--- libvirt-6.6.0/src/libxl/libxl_conf.c	2020-12-23 19:05:01.966313054 +0300
++++ libvirt-6.6.0.new/src/libxl/libxl_conf.c	2020-12-23 18:59:44.820841033 +0300
+@@ -629,6 +629,11 @@ libxlMakeDomBuildInfo(virDomainDefPtr de
+             }
+         }
+ 
++	if (def->emulator_kernel && virFileExists(def->emulator_kernel))
```

I'd prefer this to fail if file doesn't exist, instead of silently using default.

---

(same for initrd)